src/library/attribute_manager.h

```diff
@@ -7,65 +7,149 @@ Author: Leonardo de Moura
 #pragma once
 #include "kernel/environment.h"
 #include "library/io_state.h"
+#include "util/priority_queue.h"
 
 #ifndef LEAN_DEFAULT_PRIORITY
 #define LEAN_DEFAULT_PRIORITY 1000u
 #endif
 
 namespace lean {
+
+struct attr_data {
+    virtual unsigned hash() const {
+        return 0;
```

Comment:
For simplicity, even a `basic/prio_attribute` allocates an `attr_data` right now. Would you suggest changing this so that `attr_record.m_data` is an empty `unique_ptr` for those attributes (in which case I'd probably have to add that information to the serialization)? Or should I just move the `hash` implementation into a new `basic_attr_data` class?